### PR TITLE
its a hard rock life

### LINKED ISSUE #52: Load Hard Rock rooms using RoomBaht
We need to load any unplaced rooms in RoomBaht with the purchased rooms from SecretParty.
* Need the product codes from Secret party
* Run the same allocation and export, but on a local instance and not production
* export the spreadsheet out, according to existing schema

### PR DESCRIPTION
**Pull request by otakup0pe**
* import hard rock rooms / guests - closes #52 
* export hard rock "hotel report"
* handles placed/random assignment for both hotels
* `createStaffAndRooms` -> `create_rooms` / `create_staff` commands
* room metrics are dynamically generated
* new config `ROOMBAHT_ONLY_GUESTS` is a csv list of hotels to pay attention to when ingesting guests. defaults to `Ballys`
* fix for a room swap regression that came in with the hotfix :(

- [x] update deploy tooling for new commands
- [x] leverage `Room.is_dirty()` for dry-run on room updates
- [x] more more more sample data - more hard rock transfers / edge cases
- [x] ensure that the right hotel gates are in place on all api calls (not just auth related)

### CODE REVIEW

**Comment by otakup0pe** [comment]
Note that rooms from hard rock will _only_ be ingested on staging at this point. If we decide to enable it in prod, we need to update `ROOMBAHT_ONLY_GUESTS `.

File: backend/reservations/management/commands/fix_room.py [hunk 1]
```diff
@@ -7,14 +7,26 @@ class Command(BaseCommand):
...
     def handle(self, *args, **kwargs):
         if 'number' not in kwargs:
             raise CommandError("Must specify room number")
 
         room = None
+        hotel = None
+        if kwargs['hotel_name'].lower() == 'ballys':
+            hotel = 'Ballys'
+        elif kwargs['hotel_name'].lower() == 'hard rock' or \
+           kwargs['hotel_name'].lower() == 'hardrock':
```
Comment (mpesavento) [inline backend/reservations/management/commands/fix_room.py:23]: nit: indent looks off by a space, maybe just put on same line as above?

**Review by mpesavento — APPROVED** [review]
Looks good.
I'm having trouble tracking any changes in the business logic because the methods were C&Pd into new place and I dont know what stuck and what changed. Ideally, the Commands would have used the prior methods from a module, keeping the code consistent.
This works, ship it.

File: backend/reservations/management/commands/fix_room.py (hunk 1 above)
Comment (otakup0pe) [reply backend/reservations/management/commands/fix_room.py:23]: UGH, thanks emacs.

**Comment by otakup0pe** [comment]
>  Ideally, the Commands would have used the prior methods from a module, keeping the code consistent.

This is largely true. There was some refactoring as part of moving functionality into the `Room` / `Guest` model, but the business logic should have stayed the same.

File: backend/reservations/management/commands/fix_room.py (hunk 1 above)
Comment (mpesavento) [reply backend/reservations/management/commands/fix_room.py]: still off by one, in the other direction. align with line below

**Pull request merged**